lib/plugin-manager.coffee

```diff
@@ -135,33 +137,47 @@ class PluginManager
 
   # Working with autocomplete
   registerAutocompleteProviders: ->
-    if atom.packages.isPackageLoaded('autocomplete-plus')
-      atom.packages.activatePackage('autocomplete-plus')
-        .then (pkg) =>
-          @autocompleteModule = pkg.mainModule
-          @attachAutocompleteToNewEditorViews()
 
-  unregisterAutocompleteProviders: ->
-    @autocompleteSubscription?.off()
-    @autocompleteSubscription = null
+    # register a single "provider" with autocomplete; then we create one of our own CompleteProvider objects for each
+    # editor view, and requestHandler forwards requests to the appropriate object.
+    provider =
+      selector: '.source.haskell',
+      blacklist: '.source.haskell .comment'
+      requestHandler: (options) ->
+        return [] unless options?.editor?.haskellCompletionProvider
+
+        options.editor.haskellCompletionProvider.buildSuggestions()
+      #shouldn't need this
+      #dispose: ->
+      #  console.log "do something?"
+
+    # Register the provider
+    @acRegistration = atom.services.provide('autocomplete.provider', '1.0.0', {provider:provider})
```

Comment:
atom.services got deprecated in v0.177.0, so yeah... Please refer to updated autocomplete-plus API docs, as @joefitzgerald suggested.